main.go

```diff
@@ -2,45 +2,167 @@ package main
 
 import (
 	"fmt"
-	"log"
+	"net"
+	"strings"
+	"time"
+
+	retry "github.com/avast/retry-go"
 
 	"github.com/sensu-community/sensu-plugin-sdk/sensu"
-	"github.com/sensu/sensu-go/types"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
 )
 
 type HandlerConfig struct {
 	sensu.PluginConfig
-	Example string
+
+	Host string
+	Port string
+
+	// TagHost indicates whether we should add a tag named 'host' that holds the
+	// entity name.
+	TagHost bool
+
+	// Tags is a set of tags to add to the metric. They are merged with the tags
+	// coming from the event's metrics, precedence being given to the latter.
+	Tags map[string]string
+
+	// PrefixEntityName indicates whether we should prefix the metric name with
+	// the entity name.
+	PrefixEntityName bool
+
+	// Prefix is the prefix to add to the metric name.
+	Prefix string
+
+	// Retries is the number of attempts that should be made to connect to the
+	// OpenTSDB server.
+	Retries uint
+
+	// RetryDelay is the delay in seconds we should wait between connection
+	// retries.
+	RetryDelay uint
+
+	// SpaceReplacement is a string to replace potential spaces with in the
+	// entity name and the event's metric tags.
+	SpaceReplacement string
+
+	conn net.Conn
 }
 
 type ConfigOptions struct {
-	Example sensu.PluginConfigOption
+	Host             sensu.PluginConfigOption
+	Port             sensu.PluginConfigOption
+	TagHost          sensu.PluginConfigOption
+	Tags             sensu.PluginConfigOption
+	PrefixEntityName sensu.PluginConfigOption
+	Prefix           sensu.PluginConfigOption
+	Retries          sensu.PluginConfigOption
+	RetryDelay       sensu.PluginConfigOption
+	SpaceReplacement sensu.PluginConfigOption
 }
 
 var (
 	handlerConfig = HandlerConfig{
 		PluginConfig: sensu.PluginConfig{
-			Name:     "{{ .Name }}",
-			Short:    "{{ .Description }}",
-			Timeout:  10,
-			Keyspace: "sensu.io/plugins/{{ .Name }}/config",
+			Name:     "sensu-opentsdb-handler",
+			Short:    "an opentsdb handler built for use with sensu",
+			Keyspace: "sensu.io/plugins/sensu-opentsdb-handler/config",
 		},
 	}
 
 	handlerConfigOptions = ConfigOptions{
-		Example: sensu.PluginConfigOption{
-			Path:      "example",
-			Env:       "HANDLER_EXAMPLE",
-			Argument:  "example",
-			Shorthand: "e",
+		Host: sensu.PluginConfigOption{
+			Path:      "host",
+			Env:       "SENSU_OPENTSDB_HANDLER_HOST",
+			Argument:  "host",
+			Shorthand: "",
 			Default:   "",
```

Comment:
Does it not make sense to default to localhost?

Reply:
I'm indifferent; I don't think it makes more sense one way or another. If that's what the other plugins default to I can do that too!

Reply:
I just noticed you had a default port but not a default host. If it's fairly straightforward to set up an opentsdb instance on localhost, and is a common practice then I would say go for it!